Adds Atomic documentation (#58)
volusion/element-documentation: references/styling-your-block-with-css/README.md

```jsx
<div className="flex flex-wrap">
    <div className="w-100 w-60-l">{/* Your content here */}</div>
    <div className="w-100 w-40-l">{/* Your content here */}</div>
</div>
```

## Aphrodite

When the Atomic classes just aren't enough to meet your needs, you'll want to write custom styling with Aphrodite. You can view more information on styling your block with Aphrodite [here](../../how-to/style-a-block-with-aphrodite/README.md).

## `joinClasses()`

`joinClasses` is a helper function that is provided by the SDK and can be found in your block props. It is meant to join your classes together into a single string and contains some extra logic to ensure your class list is clean and free of unintended classes and whitespace.

#### It will:

-   Accept an infinite number of arguments and join them all together with a single space between each one.
-   Remove any unnecessary whitespace that might be caused before, between, or after your final classes.
-   Remove any invalid classes that might be output by bad logic (removes undefined, null, false, and additional whitespace), which is the biggest reason to use it over string literals.

> 

Reviewer: you could add a 4th bullet here about conditional classes, like you showed below with `isRequirementMet`

Author: The 3rd bullet point covers this case as a string literal would include `false` for that line.

Reviewer: The third bullet reads like it is defensively protecting you from bad code, but what I'm getting at is that this function allows you to proactively add logic inside of your className using ternary or `&&` statements. Not knowing about this feature, there is a lot of Site Designer that has a bunch of class name logic up above the JSX that could have been inlined more simply.

Author: 661e95202f4d54ebb584da417337a6b282c5b601

#### How to use it:

```jsx
<div
    className={joinClasses(
        'flex items-center w-100',
        css(
            classes.yourCustomAphroditeClassA,
            classes.yourCustomAphroditeClassB
        ),
        isRequirementMet && 'w-50-l' // isRequirementMet === false in this example
    )}
/>
```

Output:

```jsx
<div class="flex items-center w-100 dynamic_aphrodite_class"></div>
```

## Best Practices

#### Avoid Inline Styling

Sometimes inline styling is the best option for what you're trying to accomplish, but we recommend against using it when you can. Since you cannot use certain types of CSS with inline styling (media queries, pseudo selectors, etc) you end up creating yet another place that CSS needs to be tracked in your code. In the end, inline styling is fully supported, so do what works best.

#### Avoid important tags

> In order to make a site Google AMP complaint, [you are unable to use `!important` tags in your CSS](https://amp.dev/documentation/guides-and-tutorials/develop/style_and_layout/style_pages/).

Reviewer: ```suggestion
In order to make a site Google AMP complaint, [you are unable to use `!important` qualifiers in your CSS](https://amp.dev/documentation/guides-and-tutorials/develop/style_and_layout/style_pages/).
```

-   [z-index](https://github.com/volusion/element-atomic-css/blob/master/dist/styles/z-index.css)

#### All of Atomic

-   [Atomic](https://github.com/volusion/element-atomic-css/blob/master/dist/atomic.css)

### How to use Atomic?

The Atomic library is loaded anywhere your blocks are used, so there's no extra effort needed to pull it in. As far as using the classes, it's no different than referencing any other class in React. Example:

```jsx
<div className="flex flex-wrap">
    <div className="w-100 w-60-l">{/* Your content here */}</div>
    <div className="w-100 w-40-l">{/* Your content here */}</div>
</div>
```

## Aphrodite

When the Atomic classes just aren't enough to meet your needs, you'll want to write custom styling with Aphrodite. You can view more information on styling your block with Aphrodite [here](../../how-to/style-a-block-with-aphrodite/README.md).

## `joinClasses()`

`joinClasses` is a helper function that is provided by the SDK and can be found in your block props. It is meant to join your classes together into a single string and contains some extra logic to ensure your class list is clean and free of unintended classes and whitespace.

#### It will:

> -   Accept an infinite number of arguments and join them all together with a single space between each one.

Author: ```suggestion
-   Accept any number of arguments and join them all together with a single space between each one.
```
